Lib/test/test_pstats.py

```diff
@@ -33,6 +34,41 @@ def test_add(self):
         stats = pstats.Stats(stream=stream)
         stats.add(self.stats, self.stats)
 
+    def test_sort_stats_int(self):
+        valid_args = {-1: 'stdname',
+                      0: 'calls',
+                      1: 'time',
+                      2: 'cumulative'}
+        for arg_int, arg_str in valid_args.items():
+            self.stats.sort_stats(arg_int)
+            self.assertEqual(self.stats.sort_type,
+                             self.stats.sort_arg_dict_default[arg_str][-1])
+
+    def test_sort_stats_string(self):
+        for arg in SortKey:
+            arg = arg.value
+            # 'file' sorting criteria will not work because it creates
+            # ambiquity with 'filename'
+            if arg == 'file':
+                continue
```

Comment:
I don't understand -- `'file'`, `'filename'`, and `'module'` should all create the same sort -- what ambiguity is there?

Reply:
While what you're saying is essentially correct, however as it is currently implemented (without any of my changes), we're allowed to abbreviate the string argument into the `sort_stats` as long as the abbreviation is unambiguous. For example, `sort_stats` would work if either `'filename'` or `'filena'` string is passed in.  But, if string `'file'` is passed, then `sort_stats` can't differentiate between `'filename'` or `'file'` and `sort_stats` will fail with KeyError exception.   We should probably remove `'file'`.  I can do that, if you agree.  I'm guessing that no existing code is using `'file'` argument at this time.

Reply:
`file` and `filename` map to the same stat, so if we removed `file` and somebody passed in `file` it should still work.  

Remove `"file"` and write a test to make sure it still works as intended.

Reply:
Ok. Will make another pass to remove "file".